airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -527,7 +629,7 @@ def __init__(
         cluster_name: str,
         region: Optional[str] = None,
         project_id: Optional[str] = None,
-        cluster_config: Optional[Dict] = None,
+        cluster_config: Optional[Union[Dict, str]] = None,
```

Comment:
Accepting strings here might cause some issues with rendering templated values when a user wants to use a templated string, due to the above setting:

```
template_fields_renderers = {'cluster_config': 'json'}
```